protocol.md

```diff
@@ -870,6 +902,12 @@ export interface TextDocumentClientCapabilities {
 ```typescript
 interface ClientCapabilities {
 	/**
+	 * The client supports receiving the result solely through $/partialResult notifications for requests from the client to the server.
+	 * If true, the server should send all data through $/partialResult and send null as the result in the final response.
```

Comment:
Yes sorry unrelated, but important. For example in go's jsonrpc2 implementation, each notification / request gets its own goroutine. We would have to add support to the jsonrpc2 library to avoid race conditions. I assume in any language which hides the concurrency part of receiving requests/notifications this will be an issue (I would be surprised if most jsonrpc2 apis aren't designed in this way).

I want to do partialResult for references, but for hover I don't want to. Why should I be forced to always use partialResult. I guess if we get the protocol libraries to make this transparent it shouldn't matter.